ReactWindows/ReactNative.Shared/Bridge/ReactBridge.cs

```diff
@@ -147,7 +176,18 @@ private void ProcessResponse(JToken response)
                     _reactCallback.Invoke(moduleId, methodId, args);
                 };
 
-                _reactCallback.OnBatchComplete();
+                if (isEndOfBatch)
+                {
+                    // onBatchComplete will be called on the native (module) ActionQueue, but
+                    // decrementPendingJSCalls will be called sync. Be aware that the bridge may still
+                    // be processing native calls when the bridge idle signaler fires.
+                    if (localBatchHadNativeModuleCalls)
+                    {
+                        _reactCallback.OnBatchComplete();
```

Comment:
OnBatchComplete is a critical piece of infrastructure (it's how the UIManagerModule batches view operations), I'm concerned with this now being hidden behind conditionals.

Reply:
I tweaked the comments to best reflect the combination of our (existing) structural differences, and moved the setting of the state to be closer to RN Android/cxxreact. I tried inverting the guards and other things, but the code didn't end up any cleaner and had the same number of conditionals. I'm open to specific suggestions.